client/src/models/Organization.js

```diff
@@ -159,14 +159,18 @@ export default class Organization extends Model {
     return this.shortName || this.longName || this.identificationCode
   }
 
-  static toIdentificationCodeString(organization) {
-    return organization.type === Organization.TYPE.PRINCIPAL_ORG
-      ? `${organization.shortName} \\ ${
-        Settings.fields.principal.org.identificationCode?.label
-      }: ${organization.identificationCode || "Not specified"}`
-      : `${organization.shortName} ${organization.longName} ${
-        organization.identificationCode || ""
-      }`
+  toIdentificationCodeString() {
+    return Organization.toIdentificationCodeString(
+      this.shortName,
+      this.longName,
+      this.identificationCode
+    )
+  }
+
+  static toIdentificationCodeString(shortName, longName, identificationCode) {
+    return `${shortName} ${longName && `| ${longName}`} ${
+      identificationCode ? `| ${identificationCode}` : ""
+    }`
```

Comment:
This renders a trailing space if both longName and identificationCode are empty. We should also try to avoid nested template literals.